server/src/main/java/org/opensearch/plugins/PluginInfo.java

```diff
@@ -152,11 +179,16 @@ public PluginInfo(
      * @param in the stream
      * @throws IOException if an I/O exception occurred reading the plugin info from the stream
      */
+    @SuppressWarnings("unchecked")
     public PluginInfo(final StreamInput in) throws IOException {
         this.name = in.readString();
         this.description = in.readString();
         this.version = in.readString();
-        this.opensearchVersion = in.readVersion();
+        if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
+            this.opensearchVersionRanges = (List<SemverRange>) in.readGenericValue();
+        } else {
+            this.opensearchVersionRanges = List.of(new SemverRange(in.readVersion(), SemverRange.RangeOperator.DEFAULT));
+        }
```

Comment:
How will we handle this in older versions, I'm concerned that PluginInfo serialized on `v2.N-1` won't be compatible with `v2.N`.  I think the only want to handle this during the backport is to support both the existing value and a new value.

Reply:
Note; this is OK on v3.0.0 since we can introduce breaking changes, but we will need to add a deprecated noticed on the field in a 2.X release.

Reply:
Any thoughts on addressing this?

Reply:
When backporting to 2.N, the code will be updated to check for V2.N in line 187 instead of Version.V_3_0_0. 
PluginInfo serialized with V2.N-1 will only have opensearch version and not list of ranges and will be interpreted in V2.N as such in line 190. Looking to understand if the serialization issue will still persist (and in which case).

Reply:
It shouldn't really be a problem. We constantly add new fields over the wire. 
That said this PR should have 2 additional changes:
1. Merge to main with the wire contract that semver range is in 3.x only
2. Raise another PR latest 2.x to able to read and write the new field. 
3. Update the wire compatibility check on 3.x to read the field from 2.x. 

Also lets add BWC tests, this will confirm it will work in mixed cluster and rolling upgrade cases.

Reply:
@abseth-amzn Could create an issue to capture any follow up items such as this one for future PR - that will resolve this thread for me.

Reply:
https://github.com/opensearch-project/OpenSearch/issues/12008